Adjust include paths for Linux/BSD.

ED6E0F17 opened the pull request:
Include pthreads path directly on __UNIX__ . Tested that Userland still builds on Raspbian, but may  have impact on other platforms e.g. Android.

popcornmix commented:
Have you got a link that clarifies when ``__UNIX__`` is defined?
We'd like to sure it is defined for all linux/pthread platforms, but not for other supported platforms (android is the worry).

ED6E0F17 commented:
I would expect that Android IS Unix, so I have guarded against that now, although I would have to use the Linux pthreads path myself if I wanted to build Userland for Brillo.

I have also removed the extra Number Sign and squashed the commits.

__UNIX__ seems to be defined for GCC and Clang for all BSD/Linux except OSX :
http://nadeausoftware.com/articles/2012/01/c_c_tip_how_use_compiler_predefined_macros_detect_operating_system

popcornmix commented:
gcc with no options on both Ubuntu native, Ubuntu cross compiling for arm and raspbian native define ``__unix__`` and ``__linux__`` but not ``__UNIX__`` or ``__LINUX__``
As such I can't cross compile omxplayer for example with this PR.

popcornmix commented:
Try http://paste.ubuntu.com/15384596/ with ``gcc test.c``

popcornmix commented:
Also this seems to only be a partial solution.
If I remove include paths from omxplayer it just fails with other includes like:
https://github.com/raspberrypi/userland/blob/master/interface/vmcs_host/vcgencmd.h#L33

ED6E0F17 commented:
It does look bad when I am using uppercase __UNIX__ and link to a page that shows all lowercase __unix__ without noticing the difference.

"vchost_config.h" is in another path; I will see what I can do for that.

ED6E0F17 commented:
Fixed case for __unix__ and checked that __ANDROID__ is uppercase and that the  Android toolchain defines both __unix__ and __linux__

Checked that Userland still builds on Raspbian, after fixing another mistake.

Applied the same logic when including vchost_config.h

popcornmix commented:
I've been using this locally for some weeks. I've made a couple of minor tweaks and have now pushed it out. Hopefully it will be fine. Thanks.

ED6E0F17 commented:
Thanks. This only affects a handful of users every year, but it should close a few issues in the bug tracker.
